tunnel/intra/protect/protect.go

```diff
@@ -36,6 +47,34 @@ func makeControl(p Protector) (func(string, string, syscall.RawConn) error) {
 	}
 }
 
+// Returns the first IP address that is of the specified length (i.e. family).
+func scan(ips []string, l int) string {
```

Comment:
Using the address length may be error prone. Can we use `net.ParseIP(ip).To4() != nil` to determine the address family?

Reply:
OK, done.